src/Sign.Core/SignatureProviders/ClickOnceSignatureProvider.cs

```diff
@@ -80,68 +80,57 @@ public async Task SignAsync(IEnumerable<FileInfo> files, SignOptions options)
                         List<FileInfo> deployFilesToSign = filteredFiles
                             .Where(f => ".deploy".Equals(f.Extension, StringComparison.OrdinalIgnoreCase))
                             .ToList();
                         List<FileInfo> contentFiles = new();
 
-                        foreach (FileInfo dfile in deployFilesToSign)
-                        {
-                            // Rename to file without extension
-                            var dest = dfile.Replace(".deploy", "");
-                            File.Move(dfile.FullName, dest.FullName);
-                            contentFiles.Add(dest);
-                        }
+                        RemoveDeployExtension(deployFilesToSign, contentFiles);
 
                         List<FileInfo> filesToSign = contentFiles.ToList(); // copy it since we may add setup.exe
-                        var setupExe = filteredFiles.Where(f => ".exe".Equals(f.Extension, StringComparison.OrdinalIgnoreCase));
+                        IEnumerable<FileInfo> setupExe = filteredFiles.Where(f => ".exe".Equals(f.Extension, StringComparison.OrdinalIgnoreCase));
                         filesToSign.AddRange(setupExe);
 
                         // sign the inner files
                         await _aggregatingSignatureProvider.Value.SignAsync(filesToSign!, options);
 
                         // rename the rest of the deploy files since signing the manifest will need them
-                        var filesExceptFiltered = zip.GetFiles().Except(filteredFiles, FileInfoComparer.Instance).ToList();
-                        var deployFiles = filesExceptFiltered
+                        List<FileInfo> filesExceptFiltered = zip.GetFiles().Except(filteredFiles, FileInfoComparer.Instance).ToList();
+                        List<FileInfo> deployFiles = filesExceptFiltered
                             .Where(f => ".deploy".Equals(f.Extension, StringComparison.OrdinalIgnoreCase))
                             .ToList();
 
-                        foreach (var dfile in deployFiles)
-                        {
-                            // Rename to file without extension
-                            var dest = dfile.Replace(".deploy", "");
-                            File.Move(dfile.FullName, dest.FullName);
-                            contentFiles.Add(dest);
-                        }
+                        RemoveDeployExtension(deployFiles, contentFiles);
 
                         // at this point contentFiles has all deploy files renamed
 
                         // Inner files are now signed
                         // now look for the manifest file and sign that
 
-                        var manifestFile = filteredFiles.Single(f => ".manifest".Equals(f.Extension, StringComparison.OrdinalIgnoreCase));
+                        FileInfo manifestFile = filteredFiles.Single(f => ".manifest".Equals(f.Extension, StringComparison.OrdinalIgnoreCase));
```

Comment:
nit: consider an introducing some idiom in the codebase that allows succinct file extension checking. Maybe it's a new LINQ method (`SingleExtension` or `WhereExtension`) or a helper method called in existing `LINQ` methods. Or a wrapper around the extension LINQ expressions. Seems a bit verbose as-is. You need to think about the StringComparison each time but really it should be an impl. detail of a helper (IMO. Just a thought!